### PR TITLE
Move `assert.Should()` to `testkit.ToSatisfy()`.

### PR DESCRIPTION
**Pull request by jmalloc**
#### What change does this introduce?

This PR moves "user defined assertions" to the root package as "expectations".

- `assert.Should()` is now `testkit.ToSatisfy()`
- `assert.S` is now `testkit.SatisfyT`

#### What issues does this relate to?

Partially addresses #127

#### Why make this change?

The assertion package is being removed to collapse all of testing functionality into the root package.

#### Is there anything you are unsure about?

No

### CODE REVIEW

File: satisfy.go [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+
+// callerName gives the function name (qualified with a package path)
+// for the caller after skip frames (where 0 means the current function).
+func callerName(skip int) string {
+	frames := stack(skip, 1)
+	frame, _ := frames.Next()
+	return frame.Function
+}
+
+// stack returns a frames *above the caller* on the stack.
```
Comment (jmalloc) [inline satisfy.go:380]:
```suggestion
// stack returns frames *above the caller* on the stack.
```

File: satisfy.go [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+		}
+	}
+
+	return rep
+}
+
+// SatisfyT use used within expectations made via ToSatisfy() to enforce the
+// expectation.
+//
+// It is analogous the *testing.T type that is passed to tests in the native Go
```
Comment (danilvpetrov) [inline satisfy.go:111]:
```suggestion
// It is analogous to the *testing.T type that is passed to tests in the native Go
```

File: satisfy.go [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+		t.explanation = fmt.Sprintf("%s() called at %s:%d", fn, file, line)
+	} else {
+		t.explanation = fmt.Sprintf("%s() called indirectly by call at %s:%d", fn, file, line)
+	}
+}
+
+// findFrame searches, starting after skip frames, for the first caller frame
+// in a function not marked as a helper and returns that frame.
+//
+// The search stops if it finds a method of should.
```
[the remark this answers is not in the document]
Comment (jmalloc) [reply satisfy.go:345]:
```suggestion
// The search stops if it finds a frame for a method on the toSatisfy type.
```

**Review by danilvpetrov — APPROVED** [review]
Looking good, some grammar suggestions added.

File: satisfy.go [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+}
+
+// skip marks the test as failed and sets the explanation message to m.
+func (t *SatisfyT) skip(fn string) {
+	t.skipped = true
+	t.explain(fn)
+	panic(abortSentinel{})
+}
+
+// fail marks the test as failed and sets the explanation message to m.
```
Comment (jmalloc) [inline satisfy.go:311]:
```suggestion
// fail marks the test as failed. 
// fn is the name of the function that was called to indicate failure.
```

File: satisfy.go [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+	t.messages = append(t.messages, m)
+}
+
+// logf formats and adds a log message.
+func (t *SatisfyT) logf(format string, args []interface{}) {
+	m := fmt.Sprintf(format, args...)
+	t.messages = append(t.messages, m)
+}
+
+// skip marks the test as failed and sets the explanation message to m.
```
[the remark this answers is not in the document]
Comment (jmalloc) [reply satisfy.go:304]:
```suggestion
// skip marks the test as skipped. 
// fn is the name of the function that was called to skip the test.
```

File: satisfy.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (koden-km) [reply satisfy.go:345]: Maybe "where _the_ user-supplied expectation"?

[the remark this answers is not in the document]
Comment (jmalloc) [reply satisfy.go:345]: Sure, if you wanna do ENGLISH! 😄

**Review by koden-km — APPROVED** [review]

**Pull request merged**